### ISSUE TITLE
Add Local Database to add offline support in App.

### ISSUE THREAD

**Issue by ujjwalagrawal17** [WikiEducationFoundation/apps-android-wikiedudashboard#145]
**Describe the feature you'd like**

Use Room or any other library to maintain a local database to add offline support. Other suggestions for databases are welcome. 

**Would you like to work on the issue?**

No

**Comment by kiranofans** [comment]
Just asking, is Room a database like MySQL or a library that supports a database? And can I try this?

**Comment by ujjwalagrawal17** [comment]
Library that supports DB. It deals with Sqlite DB but we don't have to write SQL queries if we are using room.

**Comment by gconnect** [comment]
I would like to work on this. This issue will be broken down into sub-issues
- initial database setup 
- DAO (Database Access Object)
- Entities
-ViewModels
- Repository

**Comment by kiranofans** [comment]
> I would like to work on this. This issue will be broken down into sub-issues
> 
> * initial database setup
> * DAO (Data Access Object)
> * Entities
> * ViewModels
> * Repository

@gconnect 
Looks good to me

**Comment by jerryOkafor** [comment]
@gconnect I don't think you need `ViewModels` at this point, the app is written with MVP. Until it is migrated to MVVM, I suppose we can not add `ViewModels` to it.

**Comment by ujjwalagrawal17** [comment]
@jerryOkafor we have migrated the app from MVP to MVVM. Changes in Readme is pending. Please clone the code and check.
